cmd/policy-assistant/pkg/cli/analyze.go

```diff
@@ -296,59 +308,114 @@ func shouldIncludeANPandBANP(client *kubernetes.Clientset) (bool, bool) {
+	var allTraffic []*matcher.Traffic
+
+	if trafficPath != "" && (sourceWorkloadTraffic != "" || destinationWorkloadTraffic != "" || port != 0 || protocol != "") {
+		logrus.Fatalf("%+v", errors.Errorf("If using traffic path, you can't input traffic via CLI and viceversa"))
+	}
+
+	if trafficPath != "" {
+		allTraffics, err := json.ParseFile[[]*matcher.Traffic](trafficPath)
+		utils.DoOrDie(err)
+		for _, traffic := range *allTraffics {
+			var podA, podB *matcher.TrafficPeer
+
+			// Determine source and destination peer information
+			sourceInternal := traffic.Source.Internal
+			destinationInternal := traffic.Destination.Internal
+
+			podA = matcher.CreateTrafficPeer(traffic.Source.IP, nil)
+			podB = matcher.CreateTrafficPeer(traffic.Destination.IP, nil)
+
+			// Resolve internal peer info based on workloads
+			//sourcePeer := GetInternalPeerInfo(sourceInternal.Workload)
+			//destinationPeer := GetInternalPeerInfo(destinationInternal.Workload)
+
+			// Update podA and podB if internal information is available
+			if sourceInternal != nil {
+				podA = matcher.CreateTrafficPeer(traffic.Source.IP, &matcher.InternalPeer{
+					PodLabels:       sourceInternal.PodLabels,
+					NamespaceLabels: sourceInternal.NamespaceLabels,
+					Namespace:       sourceInternal.Namespace,
+					Workload:        sourceInternal.Workload,
+				})
+			}
+
+			if destinationInternal != nil {
+				podB = matcher.CreateTrafficPeer(traffic.Destination.IP, &matcher.InternalPeer{
+					PodLabels:       destinationInternal.PodLabels,
+					NamespaceLabels: destinationInternal.NamespaceLabels,
+					Namespace:       destinationInternal.Namespace,
+					Workload:        destinationInternal.Workload,
+				})
+			}
+
+			// Special case handling for workload-specific traffic (internal vs. external)
+			if sourceInternal != nil && destinationInternal != nil {
+				if sourceInternal.Workload != "" && destinationInternal.Workload != "" {
+					podA = matcher.GetInternalPeerInfo(sourceInternal.Workload)
+					podB = matcher.GetInternalPeerInfo(destinationInternal.Workload)
+				} else if sourceInternal.Workload != "" {
+					podA = matcher.GetInternalPeerInfo(sourceInternal.Workload)
+				} else if destinationInternal.Workload != "" {
+					podB = matcher.GetInternalPeerInfo(destinationInternal.Workload)
+				}
+			}
+
+			// Append the resolved traffic to the allTraffic slice
+			allTraffic = append(allTraffic, matcher.CreateTraffic(podA, podB, traffic.ResolvedPort, string(traffic.Protocol)))
+		}
+	} else {
+
+		if protocol != "TCP" && protocol != "UDP" && protocol != "SCTP" {
+			logrus.Fatalf("Bad Protocol Value: protocols supported are TCP, UDP and SCTP")
+		}
+
+		sourceWorkloadInfo = matcher.WorkloadStringToTrafficPeer(sourceWorkloadTraffic)
+		destinationWorkloadInfo = matcher.WorkloadStringToTrafficPeer(destinationWorkloadTraffic)
+
+		if sourceWorkloadInfo.Internal.Pods == nil || destinationWorkloadInfo.Internal.Pods == nil {
```

Comment:
if translate fails, will `Internal` be nil? Just wondering in case of nil pointer panic

Reply:
`len(pods) > 0` would be safer in case the code was ever modified to return an empty list for Pods

Reply:
Internal won´t ne nill: 
![image](https://github.com/user-attachments/assets/58ba20c6-663b-4228-97a1-092d76654e5e)

example scenario:
![image](https://github.com/user-attachments/assets/0e31e97b-5bec-4886-a290-2494e2324714)